spec/java/util/stream/DoubleStream.main.lsl

```diff
@@ -0,0 +1,1266 @@
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    @Phantom proc _find_max_loop (i: int, max: double): void
+    {
+        if (max > this.storage[i])
+            max = this.storage[i];
+    }
+
+
+    fun *.count (@target self: DoubleStream): long
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        result = this.length;
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.anyMatch (@target self: DoubleStream, predicate: DoublePredicate): boolean
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        result = false;
+
+        var i: int = 0;
+        action LOOP_WHILE(
+            i < this.length && !action CALL(predicate, [this.storage[i]]),
+            _predicate_condition_increment_loop(i)
+        );
+
+        if (i < this.length)
+            result = true;
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    @Phantom proc _predicate_condition_increment_loop (i: int): void
+    {
+        i += 1;
+    }
+
+
+    fun *.allMatch (@target self: DoubleStream, predicate: DoublePredicate): boolean
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        result = true;
+
+        if (this.length > 0)
+        {
+            result = false
+            var i: int = 0;
+            action LOOP_WHILE(
+                i < this.length && action CALL(predicate, [this.storage[i]]),
+                _predicate_condition_increment_loop(i)
+            );
+
+            if (i == this.length)
+                result = true;
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.noneMatch (@target self: DoubleStream, predicate: DoublePredicate): boolean
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        result = true;
+
+        if (this.length > 0)
+        {
+            result = false
+            var i: int = 0;
+            action LOOP_WHILE(
+                i < this.length && !action CALL(predicate, [this.storage[i]]),
+                _predicate_condition_increment_loop(i)
+            );
+
+            if (i == this.length)
+                result = true;
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.findFirst (@target self: DoubleStream): OptionalDouble
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        result = _findFirst();
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.findAny (@target self: DoubleStream): OptionalDouble
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        result = _findFirst();
+        this.linkedOrConsumed = true;
+    }
+
+
+    fun *.iterator (@target self: DoubleStream): PrimitiveIterator_OfDouble
+    {
+        result = new DoubleStreamIteratorAutomaton(state = Initialized,
```

Comment:
Does current stream should be consumed or not?

Reply:
Fixed;